optaplanner-wb-screens/optaplanner-wb-guided-rule-editor/optaplanner-wb-guided-rule-editor-client/src/main/java/org/optaplanner/workbench/screens/guidedrule/client/validation/ScoreHolderPatternBindingValidator.java

```diff
@@ -0,0 +1,36 @@
+package org.optaplanner.workbench.screens.guidedrule.client.validation;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+
+import javax.inject.Inject;
+
+import org.drools.workbench.screens.guided.rule.client.editor.validator.PatternBindingValidator;
+import org.guvnor.common.services.shared.message.Level;
+import org.guvnor.common.services.shared.validation.model.ValidationMessage;
+import org.jboss.errai.ui.client.local.spi.TranslationService;
+import org.optaplanner.workbench.screens.guidedrule.client.resources.i18n.GuidedRuleEditorConstants;
+
+/**
+ * Check whether a variable name equals to 'scoreHolder', which is a reserved name in the GRE.
+ */
+public class ScoreHolderPatternBindingValidator implements PatternBindingValidator {
+
+    private static final String SCORE_HOLDER = "scoreHolder";
+
+    private TranslationService translationService;
+
+    @Inject
+    public ScoreHolderPatternBindingValidator(final TranslationService translationService) {
+        this.translationService = translationService;
+    }
+
+    @Override
+    public Collection<ValidationMessage> validate(String variableName) {
+        if (SCORE_HOLDER.equals(variableName)) {
```

Comment:
I have like feeling, in past, I saw PR where was need for use compareTo() instead of equals() in gwt client code, but I really can not remember for more details. So please take this just as no much saying comment :).

Reply:
I don't recall such a scenario.